app/src/main/java/fr/free/nrw/commons/upload/UploadMediaDetailAdapter.java

```diff
@@ -421,28 +418,33 @@ private void setUpRecentLanguagesSection(final List<Language> recentLanguages) {
         }
 
         /**
-         * Checks if the source string contains trailing whitespace
+         * Removes any leading and trailing whitespace from the source text.
          * @param source input string
-         * @return true if contains trailing whitespace and false otherwise
+         * @return a string without leading and trailing whitespace
          */
-        public Boolean checkTrailingWhitespace(String source) {
-            int len = source.length();
-            if (len == 0) {
-                return false;
+        public String removeLeadingAndTrailingWhitespace(String source) {
+            int firstNonWhitespaceIndex = 0;
```

Comment:
Would you mind adding as a comment your excellent advice concerning `strip`? That will tip maintainers to perform the change wgen we switch SDK version. If possible write from which version the method is available. Thanks!

Reply:
I've added a comment relating to the `strip` method. I know that the method is first introduced in Java 11 but I'm not sure how android SDK environments are updated and if the JDK is updated separately, but I believe it might be Android SDK 33 which first supports Java 11.